core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/validation/ValidateRequest.java

```diff
@@ -102,13 +107,17 @@ private ValidationResult doValidate(BranchContext context, Writer index) throws
 		final List<Promise<Object>> validationPromises = Lists.newArrayList();
 		// evaluate selected rules
 		for (ValidationRule rule : rules) {
+			RemoteJob remoteJob = context.service(RemoteJob.class);
 			checkArgument(rule.getCheckType() != null, "CheckType is missing for rule " + rule.getId());
 			final ValidationRuleEvaluator evaluator = ValidationRuleEvaluator.Registry.get(rule.getType());
 			if (evaluator != null) {
 				validationPromises.add(pool.submit(rule.getCheckType(), () -> {
 					Stopwatch w = Stopwatch.createStarted();
 					
 					try {
+						if (monitor.isCanceled()) {
+							throw new OperationCanceledException(remoteJob.getName() + " has been canceled");
```

Comment:
The bigger problem here is that the exception thrown by this statement is caught by line 126 and the validation rule execution will fail instead of being reported as canceled.